src/schema.rs

```diff
@@ -495,6 +502,36 @@ impl Deserial for Type {
     }
 }
 
+impl From<TryFromIntError> for ParseError {
+    fn from(_: TryFromIntError) -> Self { ParseError::default() }
+}
+
+/// Try to convert the `len` to the provided size and serialize it.
+pub fn serial_length<W: Write>(
+    len: usize,
+    size_len: &SizeLength,
+    out: &mut W,
+) -> Result<(), W::Err> {
+    match size_len {
+        SizeLength::U8 => u8::try_from(len).unwrap_or_default().serial(out)?,
+        SizeLength::U16 => u16::try_from(len).unwrap_or_default().serial(out)?,
+        SizeLength::U32 => u32::try_from(len).unwrap_or_default().serial(out)?,
+        SizeLength::U64 => u64::try_from(len).unwrap_or_default().serial(out)?,
```

Comment:
What is the rationale for the default?

If the length is not convertible to the given value then it is better that the entire thing fails, as opposed to silently doing the wrong thing, as it is now.